flutter/lib/src/sentry_flutter.dart

```diff
@@ -35,19 +34,18 @@ mixin SentryFlutter {
       throw ArgumentError('OptionsConfiguration is required.');
     }
 
-    // It is necessary to initialize Flutter method channels so that
-    // our plugin can call into the native code.
-    WidgetsFlutterBinding.ensureInitialized();
-
     final flutterOptions = SentryFlutterOptions();
+
     // first step is to install the native integration and set default values,
     // so we are able to capture future errors.
-    _addDefaultIntegrations(
-      flutterOptions,
+    final defaultIntegrations = _createDefaultIntegrations(
```

Comment:
what's the benefit of refactoring this? they give the same result

Reply:
- One less side effect, input is unchanged.
- During the development we would also mutate the list of integrations here, this was removed again.